src/number_theory/zeta_values.lean

```diff
@@ -134,3 +147,228 @@ begin
 end
 
 end bernoulli_fourier_coeffs
+
+section bernoulli_periodized
+/-! In this section we use the above evaluations of the Fourier coefficients of Bernoulli
+polynomials, together with the theorem `has_pointwise_sum_fourier_series_of_summable` from Fourier
+theory, to obtain an explicit formula for `∑ (n:ℤ), 1 / n ^ k * fourier n x`. -/
+
+/-- The Bernoulli polynomial, extended from `[0, 1)` to the unit circle. -/
+def periodized_bernoulli (k : ℕ) : 𝕌 → ℝ := add_circle.lift_Ico 1 0 (bernoulli_fun k)
+
+lemma periodized_bernoulli.continuous {k : ℕ} (hk : k ≠ 1) : continuous (periodized_bernoulli k) :=
+add_circle.lift_Ico_zero_continuous
+  (by exact_mod_cast (bernoulli_fun_endpoints_eq_of_ne_one hk).symm)
+  (polynomial.continuous _).continuous_on
+
+lemma fourier_coeff_bernoulli_eq {k : ℕ} (hk : k ≠ 0) (n : ℤ) :
+  fourier_coeff (coe ∘ periodized_bernoulli k : 𝕌 → ℂ) n = -k! / (2 * π * I * n) ^ k :=
+begin
+  have : (coe ∘ periodized_bernoulli k : 𝕌 → ℂ) = add_circle.lift_Ico 1 0 (coe ∘ bernoulli_fun k),
+  { ext1 x, refl },
+  rw [this, fourier_coeff_lift_Ico_eq],
+  simpa only [zero_add] using bernoulli_fourier_coeff_eq hk n,
+end
+
+lemma summable_bernoulli_fourier {k : ℕ} (hk : 2 ≤ k) :
+  summable (λ n, -k! / (2 * π * I * n) ^ k : ℤ → ℂ) :=
+begin
+  have : ∀ (n : ℤ), -(k! : ℂ) / (2 * π * I * n) ^ k
+    = (-k! / (2 * π * I) ^ k) * (1 / n ^ k),
+  { intro n, rw [mul_one_div, div_div, ←mul_pow], },
+  simp_rw this,
+  apply summable.mul_left,
+  rw ←summable_norm_iff,
+  have : (λ (x : ℤ), ‖1 / (x:ℂ) ^ k‖) = (λ (x : ℤ), |1 / (x:ℝ) ^ k|),
+  { ext1 x,
+    rw [norm_eq_abs, ←complex.abs_of_real],
+    congr' 1,
+    norm_cast },
+  simp_rw this,
+  rw [summable_abs_iff],
+  exact real.summable_one_div_int_pow.mpr (one_lt_two.trans_le hk),
+end
+
+lemma has_sum_one_div_pow_mul_fourier_mul_bernoulli_fun {k : ℕ} (hk : 2 ≤ k)
+  {x : ℝ} (hx : x ∈ Icc (0:ℝ) 1) :
+  has_sum (λ n:ℤ, 1 / (n:ℂ) ^ k * fourier n (x : 𝕌)) (-(2 * π * I) ^ k / k! * bernoulli_fun k x) :=
+begin
+  -- first show it suffices to prove result for `x ∈ Ico 0 1`
+  rw [←Ico_insert_right (zero_le_one' ℝ), mem_insert_iff, or.comm] at hx,
+  have result := _,
```

Comment:
A proof with a line `have result := _` will typically be very hard to read. Can you rewrite it with a `suffices` (or with a separate lemma that you reduce to if this is more convenient)?

Reply:
In an ideal world we'd have some dedicated tactic for automating this proof pattern in a readable way (like "suffices" but changing the hypotheses rather than the goal). In fact this seems to be basically what Mario's `wlog'` tactic [zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/wlog.20tactic.20.3A.20documentation.2Fhowto.3F) does, so it's a pity that this never made it into mathlib.

Reply:
Can I get away with `suffices : ∀ {y : ℝ}, y ∈ Ico (0:ℝ) 1 → has_sum _ _,`?

Reply:
I feel that copying out the arguments of `has_sum` from 2 lines above makes the proof less readable rather than more.